__tests__/integration.test.js

```diff
@@ -308,3 +307,61 @@ describe('selectUsers()', () => {
     });
   });
 });
+
+describe('selectArticles by topic query()', () => {
+  test('200:Selecting all topics with valid topic query', () => {
+    return request(app)
+      .get('/api/articles?topic=mitch')
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles).toHaveLength(12);
+        expect(body.articles[0]).toMatchObject({
+          article_id: expect.any(Number),
+          title: expect.any(String),
+          topic: expect.any(String),
+          body: expect.any(String),
+          created_at: expect.any(String),
+          votes: expect.any(Number),
+          article_img_url: expect.any(String),
+          comment_count: expect.any(String),
+        });
+      });
+  });
+  test('200: Return [] with valid topic query but no article with this query', () => {
+    return request(app)
+      .get('/api/articles?topic=paper')
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles).toHaveLength(0);
+        expect(body.articles).toEqual([]);
+      });
+  });
+  test('200: Return all article with invalid topic query', () => {
```

Comment:
A 404 not found error would make more sense here from a user perspective for when the queried topic does not exist 🤔